src/settings/appsettings.cpp

```diff
@@ -664,6 +664,53 @@ QOnlineTts::Emotion AppSettings::defaultEmotion(QOnlineTranslator::Engine engine
     }
 }
 
+QMap<QOnlineTranslator::Language, QLocale::Country> AppSettings::regions(QOnlineTranslator::Engine engine) const
+{
+    switch (engine) {
+    case QOnlineTranslator::Google: {
+        QMap<QOnlineTranslator::Language, QLocale::Country> regions;
+        for (int i = 1; i <= QOnlineTranslator::Zulu; ++i)
+            if (!QOnlineTts::validRegions(static_cast<QOnlineTranslator::Language>(i)).isEmpty())
+                regions[static_cast<QOnlineTranslator::Language>(i)] = m_settings->value(QString("TTS/GoogleRegion/%1").arg(QOnlineTranslator::languageName(static_cast<QOnlineTranslator::Language>(i))), QLocale::AnyCountry).value<QLocale::Country>();
+        return regions;
+    }
+    case QOnlineTranslator::Bing:
+    case QOnlineTranslator::Yandex:
+    case QOnlineTranslator::LibreTranslate:
+    case QOnlineTranslator::Lingva:
+        return {};
+    default:
+        Q_UNREACHABLE();
+    }
+}
+
+void AppSettings::setRegions(QOnlineTranslator::Engine engine, const QMap<QOnlineTranslator::Language, QLocale::Country> &regions)
+{
+    switch (engine) {
+    case QOnlineTranslator::Google:
+        for (auto it = regions.cbegin(); it != regions.cend(); ++it)
+            if (!QOnlineTts::validRegions(it.key()).isEmpty())
```

Comment:
I think here we can iterate over the passed `regions` variable. This checking seem unnecessary to me.